src/motion_input_shaping/zero_vibration_stream_generator.py

```diff
@@ -0,0 +1,311 @@
+"""
+Contains the ZeroVibrationShaper class for re-use in other code.
+
+Run the file directly to test the class functionality.
+"""
+
+
+import math
+from enum import Enum
+from dataclasses import dataclass
+from typing import NamedTuple
+import numpy as np
+from plant import Plant
+
+
+class ShaperType(Enum):
+    """Enumeration for different input shaper types."""
+
+    ZV = 1
+    ZVD = 2
+    ZVDD = 3
+
+
+@dataclass
```

Comment:
Just curious, why did you choose to use `@dataclass` for `StreamSegment`, but `NamedTuple` for `AccelPoint`?

Reply:
In line 287, I set the final position of the last segment to the target position in order to remove any small rounding errors. I used `dataclass` for `StreamSegment` to make that easier since `NamedTuple` is immutable. I don't think there is a reason for `AccelPoint` to specifically be one or the other so I'll change that to `dataclass` for consistency.